src/chat.ts

```diff
@@ -134,143 +106,48 @@ export class ChatProvider implements vscode.WebviewViewProvider {
+            // Note: the command (registered in extension.ts) attaches additional workspace context.
+            console.log(request);
+            vscode.commands.executeCommand(
+              "sourcery.coding_assistant",
+              request
+            );
             break;
-          }
-          case "chat/cancelRequest": {
-            vscode.commands.executeCommand("sourcery.coding_assistant", {
-              view: "chat",
-              request: "cancel",
-            });
-            break;
-          }
-          case "recipes/recipeRequest": {
-            vscode.commands.executeCommand("sourcery.coding_assistant", {
-              view: "chat",
-              request: "sendMessage",
-              message: request,
-            });
-            break;
-          }
-          case "recipes/initialiseRequest": {
-            console.log("initialising recipes");
-            this._view.webview.postMessage({
-              command: "recipes/addRecipes",
-              result: this.recipes,
-            });
-            break;
-          }
-          case "review/initialiseRequest": {
-            console.log("initialising review");
-            this._view.webview.postMessage({
-              command: "review/addBranches",
-              result: this.branches,
-            });
-            break;
-          }
-          case "review/reviewRequest": {
-            vscode.commands.executeCommand("sourcery.coding_assistant", {
-              view: "review",
-              request: "sendMessage",
-              message: request,
-            });
-            break;
-          }
-          case "review/cancelRequest": {
-            vscode.commands.executeCommand("sourcery.coding_assistant", {
-              view: "review",
-              request: "cancel",
-            });
-            break;
-          }
-          case "review/clearRequest": {
-            this.clearReview();
-            vscode.commands.executeCommand("sourcery.coding_assistant", {
-              view: "review",
-              request: "clear",
-            });
-            break;
-          }
-          case "openLinkRequest": {
-            if (request.linkType === "url") {
-              vscode.env.openExternal(vscode.Uri.parse(request.link));
-            } else {
-              let path = vscode.Uri.file(request.link);
-
-              // Make the path relative to the workspace root
-              if (!request.link.startsWith("/")) {
-                const workspaceRoot =
-                  vscode.workspace.workspaceFolders?.[0].uri;
-                if (workspaceRoot) {
-                  path = vscode.Uri.joinPath(workspaceRoot, request.link);
-                }
+          case "extension":
+            switch (request.request) {
+              case "openLink":
+                this.handleOpenLinkRequest(request);
+                break;
+              case "insertAtCursor": {
+                this.handleInsertAtCursorRequest(request);
+                break;
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏This is now super clear and simple